test/plugin/test_filter_parser.rb

```diff
@@ -0,0 +1,730 @@
+        key_name foo
+        format json
+      ]
+    }
+    assert_nothing_raised {
+      create_driver %[
+        key_name foo
+        <parse>
+          @type ltsv
+        </parse>
+      ]
+    }
+    assert_nothing_raised {
+      create_driver %[
+        key_name message
+        suppress_parse_error_log true
+        <parse>
+          @type regexp
+          expression /^col1=(?<col1>.+) col2=(?<col2>.+)$/
+        </parse>
+      ]
+    }
+    assert_nothing_raised {
+      create_driver %[
+        key_name message
+        suppress_parse_error_log false
+        <parse>
+          @type regexp
+          expression /^col1=(?<col1>.+) col2=(?<col2>.+)$/
+        </parse>
+      ]
+    }
+    d = create_driver %[
+      key_name foo
+      <parse>
+        @type regexp
+        expression /(?<x>.)/
+      </parse>
+    ]
+    assert_false d.instance.reserve_data
+  end
+
+  # CONFIG = %[
+  #   remove_prefix test
+  #   add_prefix    parsed
+  #   key_name      message
+  #   format        /^(?<x>.)(?<y>.) (?<time>.+)$/
+  #   time_format   %Y%m%d%H%M%S
+  #   reserve_data  true
+  # ]
+  def test_filter
+    d1 = create_driver(CONFIG)
+    time = Time.parse("2012-01-02 13:14:15").to_i
+    d1.run do
+      d1.feed(@tag, time, {'message' => '12 20120402182059'})
+      d1.feed(@tag, time, {'message' => '34 20120402182100'})
+      d1.feed(@tag, time, {'message' => '56 20120402182100'})
+      d1.feed(@tag, time, {'message' => '78 20120402182101'})
+      d1.feed(@tag, time, {'message' => '90 20120402182100'})
+    end
+    filtered = d1.filtered
+    assert_equal 5, filtered.length
+
+    first = filtered[0]
+    assert_equal Time.parse("2012-04-02 18:20:59").to_i, first[0]
+    assert_equal '1', first[1]['x']
+    assert_equal '2', first[1]['y']
+    assert_equal '12 20120402182059', first[1]['message']
+
+    second = filtered[1]
+    assert_equal Time.parse("2012-04-02 18:21:00").to_i, second[0]
+    assert_equal '3', second[1]['x']
+    assert_equal '4', second[1]['y']
+
+    third = filtered[2]
+    assert_equal Time.parse("2012-04-02 18:21:00").to_i, third[0]
+    assert_equal '5', third[1]['x']
+    assert_equal '6', third[1]['y']
+
+    fourth = filtered[3]
+    assert_equal Time.parse("2012-04-02 18:21:01").to_i, fourth[0]
+    assert_equal '7', fourth[1]['x']
+    assert_equal '8', fourth[1]['y']
+
+    fifth = filtered[4]
+    assert_equal Time.parse("2012-04-02 18:21:00").to_i, fifth[0]
+    assert_equal '9', fifth[1]['x']
+    assert_equal '0', fifth[1]['y']
+
+    d2 = create_driver(%[
+      key_name data
+      format   /^(?<x>.)(?<y>.) (?<t>.+)$/
+    ])
+    time = Fluent::EventTime.from_time(@default_time) # EventTime emit test
+    d2.run do
+      d2.feed(@tag, time, {'data' => '12 20120402182059'})
+      d2.feed(@tag, time, {'data' => '34 20120402182100'})
+    end
+    filtered = d2.filtered
+    assert_equal 2, filtered.length
+
+    first = filtered[0]
+    assert_equal time, first[0]
```

Comment:
Use `assert_equal_event_time`, because `EventTime#==` compares just seconds (for compatibility reason, i guess).